tendrl/monitoring_integration/grafana/alert_dashboard.py

```diff
@@ -204,123 +201,123 @@ def create_resource_dashboard(
+                                vol_name = target["target"].split(
+                                    "volumes."
+                                )[1].split(".")[0]
+                                if resource_name == vol_name:
+                                    resource_json["dashboard"]["rows"].append(
+                                        row
+                                    )
+                                    new_resource_flag = False
+                                    break
+                            elif resource_type == "bricks":
+                                # tendrl.clusters.{cid}.nodes.{h_name}.bricks.{path}.
+                                # utilization.percent-percent_bytes
+                                hostname = resource_name.split(":")[0].split(
+                                    "|")[1].replace(".", "_")
+                                resource_name = resource_name.split(
+                                    ":", 1)[1].replace("/", "|")
+                                host = target["target"].split(
+                                    "nodes."
+                                )[1].split(".")[0]
+                                brick_path = target["target"].split(
+                                    "bricks."
+                                )[1].split(".")[0]
+                                if hostname == host and \
+                                        resource_name == brick_path:
+                                    resource_json["dashboard"]["rows"].append(
+                                        row
+                                    )
+                                    new_resource_flag = False
+                                    break
+                            elif resource_type == "hosts":
+                                # tendrl.clusters.{cid}.nodes.{host_name}.
+                                # {memory/cpu/swap}.*
+                                host_name = target["target"].split(
+                                    "nodes."
+                                )[1].split(".")[0]
+                                if resource_name == host_name:
+                                    resource_json["dashboard"]["rows"].append(
+                                        row
+                                    )
+                                    new_resource_flag = False
+                                    break
+                    elif integration_id in target["target"]:
+                        resource_json["dashboard"]["rows"].append(
+                            row
+                        )
+                        new_resource_flag = False
+                        break
+            else:
+                break
+            if most_recent_panel_id < row["panels"][-1]["id"]:
+                most_recent_panel_id = row["panels"][-1]["id"]
+        if new_resource_flag:
+            new_resource.append(resource)
+    return new_resource, resource_json, most_recent_panel_id
```

Comment:
Function name is check_duplicate and with that I feel it should return a boolean. I feel name of the function should be changed and set something like `get_distinct_resource`